File: frontend/app/lib/languages.ts
export interface LanguageOption {
	value: string;
	label: string;
}

export const LANGUAGES: LanguageOption[] = [
	{ value: "en", label: "English" },
	{ value: "uk", label: "Ukrainian" },
	{ value: "de", label: "German" },
	{ value: "fr", label: "French" },
	{ value: "es", label: "Spanish" },
	{ value: "pl", label: "Polish" },
];

export function languageLabel(value: string): string {
	return LANGUAGES.find((language) => language.value === value)?.label ?? value.toUpperCase();
}

File: backend/app/modules/languages.py
LANGUAGE_NAMES: dict[str, str] = {
    "en": "English",
    "uk": "Ukrainian",
    "de": "German",
    "fr": "French",
    "es": "Spanish",
    "pl": "Polish",
}
"""Supported language codes mapped to their English display names."""

DEFAULT_LANGUAGE = "en"
"""Fallback language code used when detection is inconclusive."""


def language_name(code: str) -> str:
    """Return the display name for a language code, falling back to the code itself."""
    return LANGUAGE_NAMES.get(code, code)
